### ISSUE TITLE
Basic dataset level metadata extraction

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad-registry#108]
- create a model, let's call it `URLMetadata`, which one-to-many relationship to `URL.id` where we would extract/keep dataset level metadata
- columns to have
  - `id`
  - `url_id`: URL.id
  - `url_describe`: `str` -- copied describe from `URL` record so we could compare to see if we extracted for the same version or not
  - `extractor_name` : `str`
  - `extractor_version` : ?
  - `json_blob`: `str`

extract it for few basic extractors:
- `metalad_core`
- `datacite_gin`: from `datalad-catalog` extension: if there is `datacite.yml` (24 datasets)
- `studyminimeta` - conditionally on presence of `.studyminimeta.yaml` (there is 8 datasets which we cloned already with that file present)

- run job to extract metadata for datasets which do not have any metadata extracted yet
- add to the table  visualizing present datasets in the registry a column with which extractors have metadata extracted for that dataset

**Comment by candleindark** [comment]
Update: This issue is to be solved by https://github.com/datalad/datalad-registry/pull/113. 

At this point the solution provided by https://github.com/datalad/datalad-registry/pull/113 runs the extractor `metalad_core` on each registered dataset records the extracted data into the database, provides links to extracted data on the webUI, and provide API endpoints to the extracted data. The remaining steps to resolve this issue are:
 

- [ ] 1. Make the aforementioned behaviors available to the `datacite_gin` and `studyminimeta` extractors as well
- [ ] 2. Provide logic to  skip the metadata extraction for a particular extractor if such metadata are already available in the database. 
- [ ] 3. Provide tests to maintain same level of code coverage

**Comment by yarikoptic** [comment]
Great formalization of the next steps. Little clarification for `2.` -- skip if extractor and dataset version on which metadata was extracted are already available.  I.e. we still would want to re-extract if e.g. dataset got updated.

**Comment by candleindark** [comment]
> Great formalization of the next steps. Little clarification for `2.` -- skip if extractor and dataset version on which metadata was extracted are already available. I.e. we still would want to re-extract if e.g. dataset got updated.

I case of an update. Do we only keep the latest metadata of a dataset for a particular extractor?

**Comment by yarikoptic** [comment]
yes, I think we should care only about the latest and greatest, ATM at least.
